### PR TITLE
Bach[Athena]: Support Timedelta

### PR DESCRIPTION
**Pull request by KathiaBarahona**
Same approach as time series, we use float values (total seconds) to represent time deltas.

### CODE REVIEW

File: bach/bach/series/series_datetime.py [hunk 1]
```diff
@@ -606,10 +638,19 @@ class SeriesTimedelta(SeriesAbstractDateTime):
...
+        # None here for Athena, because it doesn't have a interval type.
+        DBDialect.ATHENA: None,
     }
     supported_value_types = (datetime.timedelta, numpy.timedelta64, str)
     supported_source_dtypes = ('string',)
 
+    @classmethod
+    def get_db_dtype(cls, dialect: Dialect) -> Optional[str]:
+        if is_athena(dialect):
+            from bach.series import SeriesString
```
Comment (KathiaBarahona) [inline bach/bach/series/series_datetime.py:650]: oops
```suggestion
```

File: bach/bach/series/series_datetime.py [hunk 2]
```diff
@@ -785,6 +837,8 @@ def quantile(
             )
 
         result = calculate_quantiles(series=self.dt.total_seconds, partition=partition, q=q)
+        if is_athena(self.engine):
+            return result.astype('timedelta').copy_override_type(SeriesTimedelta)
```
Comment (KathiaBarahona) [inline bach/bach/series/series_datetime.py:841]: Actually quantiles are not working yet, will open an issue for this
```suggestion

```

File: bach/bach/series/series_datetime.py [hunk 3]
```diff
@@ -686,10 +718,18 @@ class SeriesTimedelta(SeriesAbstractDateTime):
     supported_db_dtype = {
         DBDialect.POSTGRES: 'interval',
         DBDialect.BIGQUERY: 'INTERVAL',
+        # None here for Athena, because it doesn't have a interval type.
```
Comment (thijs-obj) [inline bach/bach/series/series_datetime.py:721]: I think we should have some more explanation here.
1.The documentation [1] contradicts this comment and does list interval types.
2. I'd expect a comment somewhere explaining how we solved this by representing intervals with floats.

[1] https://prestodb.io/docs/0.217/language/types.html#date-and-time

**Review by thijs-obj — COMMENTED** [review]
Cool solution. A lot less code than I thought.

Did leave some comments

File: bach/bach/series/series_datetime.py (hunk 3 above)
Comment (KathiaBarahona) [reply bach/bach/series/series_datetime.py]: added here https://github.com/objectiv/objectiv-analytics/pull/1168/commits/6ee5eaba7859ded752f6f2e823074d807df89f42

File: bach/tests/unit/bach/test_series_timedelta.py [hunk 4]
```diff
@@ -4,12 +4,43 @@
...
+    assert_call(timedelta64(50_000, 'D') + timedelta64(123, 's') + timedelta64(9, 'us'),  4320000123.000009)
+
+    # Special cases: Not-a-Time will be represented as NULL, and NULL itself
+    nat = timedelta64('NaT')
+    dtype = SeriesTimedelta.dtype
+    assert SeriesTimedelta.supported_value_to_literal(dialect, nat, dtype=dtype) == Expression.construct('NULL')
+    assert SeriesTimedelta.supported_value_to_literal(dialect, None, dtype=dtype) == Expression.construct('NULL')
+
+
 def test_supported_value_to_literal(dialect):
```
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/tests/unit/bach/test_series_timedelta.py:44]: will do here https://github.com/objectiv/objectiv-analytics/pull/1195

File: bach/bach/series/series_datetime.py [hunk 5]
```diff
@@ -686,10 +718,18 @@ class SeriesTimedelta(SeriesAbstractDateTime):
...
+        DBDialect.ATHENA: None,
```
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/bach/series/series_datetime.py:724]: fixed here https://github.com/objectiv/objectiv-analytics/pull/1168/commits/6ee5eaba7859ded752f6f2e823074d807df89f42

File: bach/tests/functional/bach/test_series_timedelta.py [hunk 6]
```diff
@@ -92,6 +94,10 @@ def test_timedelta(engine):
         use_to_pandas=True,
     )
 
+    if is_athena(engine):
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_series_timedelta.py:97]: **Question:** Do we have some way to track that we still need to do this? Is there an issue, or another test that will not work until we fix `mode` ?

Sorry forgot this in the first review round.

**Review by thijs-obj — COMMENTED** [review]
OK for me :+1:

File: bach/tests/functional/bach/test_series_timedelta.py (hunk 6 above)
Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_series_timedelta.py:97]: I need to create an issue. Quite sure is not working

Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_series_timedelta.py:97]: https://github.com/objectiv/objectiv-analytics/issues/1216

**Pull request merged**